src/core/lib/security/credentials/external/pluggable_auth_external_account_credentials.cc

```diff
@@ -17,20 +17,89 @@
 
 #include "src/core/lib/security/credentials/external/pluggable_auth_external_account_credentials.h"
 
+#include <cxxabi.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <algorithm>
+#include <chrono>
+#include <future>
 #include <initializer_list>
 #include <map>
+#include <memory>
+#include <thread>
 #include <utility>
 
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
 #include "absl/strings/numbers.h"
 #include "absl/strings/str_format.h"
+#include "absl/strings/str_split.h"
+#include "absl/strings/string_view.h"
+#include "absl/strings/strip.h"
 
+#include <grpc/slice.h>
+#include <grpc/support/alloc.h>
 #include <grpc/support/json.h>
+#include <grpc/support/time.h>
 
+#include "src/core/lib/gpr/subprocess.h"
+#include "src/core/lib/iomgr/load_file.h"
 #include "src/core/lib/json/json.h"
+#include "src/core/lib/json/json_reader.h"
+#include "src/core/lib/security/credentials/external/external_account_credentials.h"
+#include "src/core/lib/slice/slice.h"
+#include "src/core/lib/slice/slice_internal.h"
+
+extern char** environ;
 
 #define DEFAULT_EXECUTABLE_TIMEOUT_MS 30000  // 30 seconds
 #define MIN_EXECUTABLE_TIMEOUT_MS 5000       // 5 seconds
 #define MAX_EXECUTABLE_TIMEOUT_MS 120000     // 120 seconds
+#define SAML_SUBJECT_TOKEN_TYPE "urn:ietf:params:oauth:token-type:saml2"
+#define GOOGLE_EXTERNAL_ACCOUNT_ALLOW_EXECUTABLES \
+  "GOOGLE_EXTERNAL_ACCOUNT_ALLOW_EXECUTABLES"
+#define GOOGLE_EXTERNAL_ACCOUNT_ALLOW_EXECUTABLES_ACCEPTED_VALUE "1"
+
+char* portable_str_dup(const char* s) {
```

Comment:
Don't think this is necessarily an issue, but just want to double check that these basic functions aren't present somewhere in the library already

Reply:
Thanks for the tip, found a str_dup method already implemented, so I'll use that.
The other two methods have no implementation and it will be nice to have them in gpr/json_util. 
Should I do a PR for this?

Reply:
yeah thats a great idea, that would be a good stand alone pr